[govee] discovery fails

Version:     4.2.0 (Build)

```
2024-07-19 09:46:03.958 [DEBUG] [govee.internal.GoveeDiscoveryService] - starting Scan
2024-07-19 09:46:03.963 [DEBUG] [govee.internal.GoveeDiscoveryService] - Discovering Govee devices on name:eth0 (eth0) ...
2024-07-19 09:46:03.966 [WARN ] [.govee.internal.CommunicationManager] - exception when receiving status packet
java.net.SocketException: Socket closed
        at sun.nio.ch.DatagramSocketAdaptor.receive(DatagramSocketAdaptor.java:252) ~[?:?]
        at java.net.DatagramSocket.receive(DatagramSocket.java:710) ~[?:?]
        at org.openhab.binding.govee.internal.CommunicationManager$StatusReceiver.run(CommunicationManager.java:202) [bundleFile:?]
Caused by: java.nio.channels.AsynchronousCloseException
        at sun.nio.ch.DatagramChannelImpl.endRead(DatagramChannelImpl.java:523) ~[?:?]
        at sun.nio.ch.DatagramChannelImpl.trustedBlockingReceive(DatagramChannelImpl.java:677) ~[?:?]
        at sun.nio.ch.DatagramChannelImpl.blockingReceive(DatagramChannelImpl.java:635) ~[?:?]
        at sun.nio.ch.DatagramSocketAdaptor.receive(DatagramSocketAdaptor.java:240) ~[?:?]
        ... 2 more
2024-07-19 09:46:03.969 [DEBUG] [govee.internal.GoveeDiscoveryService] - Discovery with IO exception: Invalid argument
```

--- a/bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/CommunicationManager.java
+++ b/bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/CommunicationManager.java
@@ -202,8 +202,8 @@ public class CommunicationManager {
                         DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
                         if (!socket.isClosed()) {
                             socket.receive(packet);
+                        } else {
                             logger.warn("Socket was unexpectedly closed");
-                            // break will in turn reopen the port
                             break;
                         }
                         if (stopped) {

--- a/bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/CommunicationManager.java
+++ b/bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/CommunicationManager.java
@@ -200,7 +200,12 @@ public class CommunicationManager {
                     socket.setReuseAddress(true);
                     while (!stopped) {
                         DatagramPacket packet = new DatagramPacket(buffer, buffer.length);
-                        socket.receive(packet);
+                        if (!socket.isClosed()) {
+                            socket.receive(packet);
+                            logger.warn("Socket was unexpectedly closed");
+                            // break will in turn reopen the port
+                            break;
+                        }
                         if (stopped) {
                             break;
                         }

